### ISSUE TITLE
`DateTime` is not properly parsed when writing a Snowflake table

### ISSUE THREAD

**Issue by krish-adi** [modelstar-labs/modelstar#3]
When writing a DataFrame result from a `CALL SPROC`. The return DataFrame if it has a DateTIme column is not parsed properly and is written as a SQL Number type. This could possibly be solved by converting it first into a Snowflake_DF and then writing it to the Table. 

```
# Convert the Pandas dataframe into a Snowflake dataframe
df_sf = snowpark_session.create_dataframe(df_pd)  
  
# Write the results of the dataframe into a target table
df_sf.write.mode("overwrite").save_as_table(destination_table)
```

**Comment by krish-adi** [comment]
**Notes**
1. session.create_dataframe : creates a table first using write_pandas internally and makes the table then available as a SF_DF. So, doesn't help to solve our problem. 
2. Type casting from NUMBER/FLOAT to DATE/TIME is not possible. But, VARCHAR to DATE/TIME is possible from typecasting: https://docs.snowflake.com/en/sql-reference/data-type-conversion.html 
3. The available datetime types are: https://docs.snowflake.com/en/sql-reference/data-types-datetime.html#date-time-data-types 

**Solution**
1. Check if a column in the output dataframe is a date/time and then convert it into a string object in the dataframe and then write it to the table. 
2. Then a type conversion can be used to turn the varchar to a date or a time. https://docs.snowflake.com/en/sql-reference/data-type-conversion.html

**Comment by krish-adi** [comment]
There are type casting functions provided in Snowflake such as: 

```
select to_date('2022-04-01 01:02:03.004');
``` 

which returns 

```
TO_DATE('2022-04-01 01:02:03.004')
2022-04-01
```

**Comment by krish-adi** [comment]
In Pandas, converting a datetime column to a string column can be done using: 

```
df['ds'] =  df['ds'].dt.strftime('%Y-%m-%d %H:%M:%S')
```

**Comment by krish-adi** [comment]
Resolved in V 0.3.1.

The column with date-time is converted to a string and stored with format: '%Y-%m-%d %H:%M:%S'
